src/it/java/org/checkstyle/suppressionxpathfilter/AbstractXpathTestSupport.java

```diff
@@ -52,8 +52,8 @@
     private static final Pattern LINE_COLUMN_NUMBER_REGEX =
             Pattern.compile("([0-9]+):([0-9]+):");
 
-    @Rule
-    public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+    @TempDir
```

Comment:
`@TempDir` wont work for extensions. The field `temporaryFolder` will be `null`.
See https://stackoverflow.com/questions/54647577/how-to-use-tempdir-in-extensions for similar case.

This may be the root of the problem. I'll try to fix it.